substrate/frame/referenda/src/migration.rs

```diff
@@ -137,7 +154,7 @@ pub mod v1 {
 				if let Some(new_value) = maybe_new_value {
 					weight.saturating_accrue(T::DbWeight::get().reads_writes(1, 1));
 					log::info!(target: TARGET, "migrating referendum #{:?}", &key);
-					ReferendumInfoFor::<T, I>::insert(key, new_value);
+					v1::ReferendumInfoFor::<T, I>::insert(key, new_value);
```

Comment:
This would write to the current correct storage.
```suggestion
					ReferendumInfoFor::<T, I>::insert(key, new_value);
```

Reply:
are we not supposed to move from v0 to v1 then to this MigrateBlockNumberProvider?

Reply:
I don't think so, the change of block number provider is independent of the pallet version.
You can switch the block number provider at any time. You can also never switch it by keeping a block number provider same as before.

Reply:
I guess writing `v1` is also good, it is moving from v0 to v1.